File: source/downloads/code/LeetCode/Python/Binary-Tree-Paths.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param {TreeNode} root
    # @return {string[]}
   
    def dfs(self,ret,path,node):
        path += str(node.val)
        if node.left == None and node.right == None:
            ret.append(path[:])
        path += "->"
        if node.left != None:
            self.dfs(ret,path,node.left)
        if node.right != None:
            self.dfs(ret,path,node.right)


    def binaryTreePaths(self, root):
        ret = []
        if root == None:
            return ret
        self.dfs(ret,"",root)
        return ret

File: downloads/code/LeetCode/Java/Binary-Tree-Paths.java
/**
 * definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<String> binaryTreePaths(TreeNode root) {
        List<String> ret = new LinkedList<String>();
        if (root==null)
            return ret;
        searchBT(root,""+root.val,ret);
        return ret;    
    }

    public void searchBT(TreeNode node, String path, List<String> ret){
        if (node.left==null && node.right == null)
            ret.add(path);
        if (node.left != null)
            searchBT(node.left, path+"->"+node.left.val, ret);
        if (node.right != null)
            searchBT(node.right, path+"->"+node.right.val, ret);

    }
}
